crates/sc-consensus-subspace/src/lib.rs

```diff
@@ -78,7 +78,8 @@ use std::marker::PhantomData;
 use std::num::NonZeroUsize;
 use std::pin::Pin;
 use std::sync::Arc;
-use subspace_archiving::archiver::{ArchivedSegment, Archiver};
+pub use subspace_archiving::archiver::ArchivedSegment;
```

Comment:
I agree with you in general. However, this crate exports `ArchivedSegmentNotification` which has a public field- `ArchivedSegment`. It makes it a part of the crate's interface. Exporting this type makes the API consistent IMHO.

Reply:
I would like to prevent messy multi-layer re-exports if possible. If we do this then `subspace-service` should be re-exporting it as well, because it also appears in its public interface and a few other places.

So far we were not doing that, in fact `subspace-service` doesn't even have `subspace-archiving` in its direct dependencies.

If `subspace-service` exports it then it needs to be exposed from SDK, so the chain gets really deep.

Another concern with re-exports is that they are often forgotten long after they are needed due to clippy not complaining about it.